include/ginkgo/core/solver/solver_base.hpp

```diff
@@ -590,6 +590,14 @@ class EnablePreconditionedIterativeSolver
 };
 
 
+/**
+ * A helper for algorithm selection in the triangular solvers.
+ * It currently only matters for the Cuda executor as there,
+ * we have a choice between the Ginkgo and cuSPARSE implementations.
+ */
+enum class trisolve_algorithm { sparselib, ginkgo };
```

Comment:
I think we can be more explicit about what the algorithm does. May add a `level` algorithm later?
```suggestion
enum class trisolve_algorithm { sparselib, syncfree };
```

Reply:
Is the syncfree only on certain architecture like Volta and Ampere?

Reply:
no, it also works (with a slightly different kernel) on Pascal etc. This is an algorithmic difference: Fine-grained parallelism or level parallelism